app/alias_utils.py

```diff
@@ -459,3 +465,14 @@ def transfer_alias(alias, new_user, new_mailboxes: [Mailbox]):
     alias.pinned = False
 
     Session.commit()
+
+
+def change_alias_status(alias: Alias, enabled: bool, commit: bool = True):
+    alias.enabled = enabled
+    if commit:
+        Session.commit()
+
+    event = AliasStatusChange(
+        alias_id=alias.id, alias_email=alias.email, enabled=enabled
+    )
+    EventDispatcher.send_event(alias.user, EventContent(alias_status_change=event))
```

Comment:
I would put the commit after the send_event. This way the modification and the event happen in a single Tx